src/python_inspector/setup_py_live_eval.py

```diff
@@ -54,11 +53,24 @@ def iter_requirements(level, extras, setup_file):
     setup_requires = {}
     # change directory to setup.py path
     with pushd(os.path.dirname(setup_file)):
-        with mock.patch.object(setuptools, "setup") as mock_setup:
-            sys.path.append(os.path.dirname(setup_file))
-            g = {"__file__": setup_file, "__name__": "__main__"}
-            with open(setup_file) as sf:
-                exec(sf.read(), g)
+        with open(setup_file) as sf:
+            file_contents = sf.read()
+            setup_provider = re.findall(r"from ([a-z._]+) import setup", file_contents)
```

Comment:
There is another pattern for "import setuptools" or "import distutils" and then call the qualified setup() function. May be we would be better off to parse the file and get this from the AST?

See https://github.com/search?q="import+setuptools"+"setuptools.setup"+filename%3Asetup.py+language%3APython+language%3APython&type=Code&ref=advsearch&l=Python&l=Python

Reply:
done